obal/data/playbooks/nightly/nightly.yaml

```diff
@@ -13,21 +13,52 @@
       when: nightly_releaser is defined and '--arg jenkins_job=' in (build_package_tito_releaser_args | join(' '))
 
     - block:
-      - name: 'set nightly_macro_args'
-        set_fact:
-          nightly_macro_args:
-            - "--arg rpmbuild_options=\"--define 'nightly .{{ lookup('pipe', 'date +%Y%m%d%H%M%S') }}git{{ nightly_githash[:7] }}'\""
-
-      - name: 'append nightly_macro_args to build_package_tito_releaser_args'
-        set_fact:
-          build_package_tito_releaser_args: "{{ build_package_tito_releaser_args }} + {{ nightly_macro_args }}"
-
-      - name: copy source files
-        copy:
-          src: "{{ item }}"
-          dest: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}/"
-        with_items: "{{ nightly_sourcefiles }}"
+        - name: 'set package_dir'
+          set_fact:
+            package_dir: "{{ inventory_dir }}/{{ package_base_dir }}/{{ inventory_hostname }}/"
+
+        - name: 'set global_nightly_macro'
+          set_fact:
+            global_nightly_macro: "%global nightly .{{ lookup('pipe', 'date +%Y%m%d%H%M%S') }}git{{ nightly_githash[:7] }}"
+
+        - name: 'copy source files'
+          copy:
+            src: "{{ item }}"
+            dest: "{{ package_dir }}"
+          with_items: "{{ nightly_sourcefiles }}"
+
+        - name: 'get nightly_specfile'
+          find:
+            pattern: "*.spec"
+            path: "{{ package_dir }}"
+          register: nightly_specfile
+
+        - name: 'set nightly_specfile_path'
+          set_fact:
+            nightly_specfile_path: "{{ nightly_specfile.files[0].path }}"
+
+        - name: 'add nightly macro to specfile'
+          lineinfile:
+            path: "{{ nightly_specfile_path }}"
+            line: "{{ global_nightly_macro }}"
+            insertbefore: BOF
+            backup: yes
+          register: original_specfile
       when: nightly_sourcefiles is defined and nightly_githash is defined
-    # end block
+
   roles:
     - build_package
+
+  post_tasks:
+    - block:
+        - name: 'restore spec file'
+          copy:
+            src: "{{ original_specfile.backup }}"
+            dest: "{{ nightly_specfile_path }}"
+
+        - name: 'delete copied source files'
+          file:
+            state: absent
+            path: "{{ package_dir}}/{{ item | basename }}"
+          with_items: "{{ nightly_sourcefiles }}"
+      when: nightly_sourcefiles is defined and nightly_githash is defined
```

Comment:
I think this is redundant if you add `when: original_specfile is defined` to the `copy` block and `when: nightly_sourcefiles is defined` to the `file` block.

Reply:
I added this because I think it makes the script more reliable. Perhaps this also needs to be in an always block so we always clean up.

Reply:
Ah, I follow you. that is clearer. I quickly copied the conditional from above because I wanted them to trigger on the same conditions. But your suggestion is clearer.